### PR TITLE
Implement AdminChat Module and Command

### PR DESCRIPTION
**Pull request by mgejay**
Solves #137

### CODE REVIEW

File: src/com/superiornetworks/icarus/modules/AdminChat.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    }
+
+    public static void AdminChatMsg(CommandSender sender, String message)
+    {
+        for (Player onlineadmins : Bukkit.getOnlinePlayers())
+        {
+            if (ICM_Rank.isRankOrHigher(onlineadmins, ICM_Rank.Rank.SUPER))
+            {
+                // Some strange ass colour codes being used here...
+                onlineadmins.sendMessage("§f[§bADMIN§f] §c" + sender.getName() + ICM_Rank.getRank(sender).actag + "§f: §b " + message);
```
Comment (Wild1145) [inline src/com/superiornetworks/icarus/modules/AdminChat.java]: Could you change this to get the senders nickname? I have grown to prefer that to just their username.

**Comment by Wild1145** [comment]
This currently is not following the guidelines, it requires formatting changes and shouldn't actually work...

**Comment by mgejay** [comment]
@Wild1145 You obviously didn't really test this. I compile and test all my changes/PRs, but I will change the senders nickname.

**Comment by Wild1145** [comment]
@Adaptified No, I have not tested it. I looked and didn't think it would compile, I have got my plugins confused so it actually probably should compile.

File: src/com/superiornetworks/icarus/commands/Command_o.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+import org.apache.commons.lang.StringUtils;
+import org.bukkit.command.Command;
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Player;
+
+@CommandParameters(name = "o", description = "Toggle or use adminchat.", usage = "/o [message]", rank = ICM_Rank.Rank.SUPER)
+public class Command_o {
+
+    public boolean onCommand(CommandSender sender, Command cmd, String label, String[] args) {
+        if (args.length == 0) {
```
Comment (Wild1145) [inline src/com/superiornetworks/icarus/commands/Command_o.java]: Formatting Issues

File: src/com/superiornetworks/icarus/commands/Command_o.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+            try {
+                if (!(sender instanceof Player)) {
+                    return true;
+                }
+                Player player = (Player) sender;
+                if (!ICM_SqlHandler.inAC(player.getName())) {
```
Comment (Wild1145) [inline src/com/superiornetworks/icarus/commands/Command_o.java]: Formatting Issues

File: src/com/superiornetworks/icarus/commands/Command_o.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+                    playerMsg(sender, "&7Toggling adminchat: On");
+                    Connection c = ICM_SqlHandler.getConnection();
+                    PreparedStatement statement = c.prepareStatement("UPDATE `players` SET `inAC` = TRUE WHERE `playerName` = ?");
+                    statement.setString(1, player.getName());
+                    statement.executeUpdate();
+                } else {
```
Comment (Wild1145) [inline src/com/superiornetworks/icarus/commands/Command_o.java]: Formatting Issues

File: src/com/superiornetworks/icarus/commands/Command_o.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+                    statement.executeUpdate();
+                }
+                return true;
+            } catch (SQLException ex) {
+                Logger.getLogger(Command_o.class.getName()).log(Level.SEVERE, null, ex);
+            }
+            return true;
+        }
+
+        if (args.length >= 1) {
```
Comment (Wild1145) [inline src/com/superiornetworks/icarus/commands/Command_o.java]: Formatting Issues

**Comment by mgejay** [comment]
All done @Wild1145

File: src/com/superiornetworks/icarus/commands/Command_o.java [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+                {
+                    playerMsg(sender, "&7Toggling adminchat: Off");
+                    Connection c = ICM_SqlHandler.getConnection();
+                    PreparedStatement statement = c.prepareStatement("UPDATE `players` SET `inAC` = FALSE WHERE `playerName` = ?");
+                    statement.setString(1, player.getName());
+                    statement.executeUpdate();
+                }
+                return true;
+              } 
+                catch (SQLException ex)
```
Comment (Wild1145) [inline src/com/superiornetworks/icarus/commands/Command_o.java:49]: Formattings out here.

**Review by Wild1145 — CHANGES_REQUESTED** [review]
Additional changes are required, this will not compile.

File: src/com/superiornetworks/icarus/modules/AdminChat.java [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+    }
+
+    public static void AdminChatMsg(CommandSender sender, String message)
+    {
+        for (Player onlineadmins : Bukkit.getOnlinePlayers())
+        {
+            if (ICM_Rank.isRankOrHigher(onlineadmins, ICM_Rank.Rank.SUPER))
+            {
+                // Some strange ass colour codes being used here...
+                onlineadmins.sendMessage("§f[§bADMIN§f] §c" + ICM_SqlHandler.getNick(sender) + ICM_Rank.getRank(sender).actag + "§f: §b " + message);
```
Comment (Wild1145) [inline src/com/superiornetworks/icarus/modules/AdminChat.java:25]: This does not work. Please review this.

**Review by Wild1145 — CHANGES_REQUESTED** [review]
Please make the required changes.

**Pull request closed without merging**